shell/platform/darwin/ios/framework/Source/FlutterUndoManagerPlugin.mm

```diff
@@ -104,9 +104,11 @@ - (void)setUndoState:(NSDictionary*)dictionary API_AVAILABLE(ios(9.0)) {
     // This is needed to notify the iPadOS keyboard that it needs to update the
     // state of the UIBarButtons. Otherwise, the state changes to NSUndoManager
     // will not show up until the next keystroke (or other trigger).
-    id<UITextInputDelegate> inputDelegate =
-        _viewController.engine.textInputPlugin.textInputView.inputDelegate;
-    [inputDelegate selectionDidChange:_viewController.engine.textInputPlugin.textInputView];
+    UITextInputAssistantItem* assistantItem =
+        _viewController.engine.textInputPlugin.textInputView.inputAssistantItem;
+    NSArray<UIBarButtonItemGroup*>* leadingGroups = assistantItem.leadingBarButtonGroups;
+    assistantItem.leadingBarButtonGroups = @[];
+    assistantItem.leadingBarButtonGroups = leadingGroups;
```

Comment:
why is it setting to empty, and then setting back? Some comments here will be helpful.

Reply:
This was a workaround to notify the iPadOS keyboard to update the state of these buttons. It's been awhile, but I believe the behavior I was seeing was that state changes to canUndo/canRedo would not show up until the next keypress – so if I press a letter, the undo button should light up, but it won't until I press another key.

Reply:
Worth trying if it works if we simply do `assistantItem.leandingBarButtonGroups = assistantItem.leandingBarButtonGroups`, in case apple doesn't check for no-ops in the setter.

Reply:
BTW if it's iOS 17 related, feel free to contribute to [the design doc](https://docs.google.com/document/d/1sM3HMv-SQin39yX1aPUU7vtGv7Hcef1Quc3QhRXBl6A/edit?resourcekey=0-SFYD8vmOIkXiXCZvB1Wlcw) which already discusses about several iOS 17 text input issues. It can be a good reference doc for future developers.

Reply:
the original issue is reproduced on an iOS 16 device so probably not ios specific. 

I tested removing the intermediate empty array step it appears the setter indeed doesn't skip even when given the same array instance. Updated. I'll file a radar later since the keyboard could just rely on notifications to properly update the undo/redo UI.